build.zig

```diff
@@ -44,6 +45,24 @@ pub fn build(b: *std.Build) void {
     }
     run_step.dependOn(&run_cmd.step);
 
+    // Add snapshot tool
+    const snapshot_exe = b.addExecutable(.{
+        .name = "snapshot",
+        .root_source_file = b.path("src/snapshot.zig"),
+        .target = target,
+        .optimize = optimize,
+        .link_libc = true,
+    });
+    b.installArtifact(snapshot_exe);
+
+    // Add a step to run the snapshot tool
+    const run_snapshot = b.addRunArtifact(snapshot_exe);
+    run_snapshot.step.dependOn(b.getInstallStep());
```

Comment:
This line should not be need. `addRunArtifact` should ad all required dependencies
```suggestion
```